plugins/woocommerce-blocks/assets/js/editor-components/sidebar-compatibility-notice/index.tsx

```diff
@@ -33,6 +42,9 @@ export const CartCheckoutSidebarCompatibilityNotice = ( {
 		}
 	);
 
+	const blockName =
+		block === 'cart' ? 'woocommerce/cart' : 'woocommerce/checkout';
```

Comment:
wouldn't it be safer to concatenate "woocommerce/" with the block name an let it error out if something invalid appears here?

This way if a invalid block slips into here, it will assume woocommerce/checkout and be silent about the invalid data.

Reply:
`block` are props of the `<CartCheckoutSidebarCompatibilityNotice>` component. The only place, we're using this component is on https://github.com/woocommerce/woocommerce/blob/2749cebf7f06b5dbfd45550a8dc4f65038ff9bb8/plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/sidebar-notices/index.tsx#L133-L135

Technically, it should not be possible that `block` has a value apart from `cart` or `checkout`. Also, the following part, which is located before the part I mentioned before, should prevent that anything other than the Cart or the Checkout block every reaches this section: https://github.com/woocommerce/woocommerce/blob/2749cebf7f06b5dbfd45550a8dc4f65038ff9bb8/plugins/woocommerce-blocks/assets/js/blocks/cart-checkout-shared/sidebar-notices/index.tsx#L109-L115

That said, may I ask what scenario you have in mind? How do you think an invalid block might slip into here?

Reply:
I'm not saying it can have a value not in [cart, checkout] right now, but in the future a new value is introduced, the ternary will erroneously default to woocommerce/checkout with probably no immediate and visible side effects.

Reply:
I see what you mean, @wavvves. I believe that if we introduce a new value in the future, the TS types should be adjusted accordingly. The part, that you pointed out is located at

https://github.com/woocommerce/woocommerce/blob/27899be1549d2d3670ca1e7527a5a88fbbc499c8/plugins/woocommerce-blocks/assets/js/editor-components/sidebar-compatibility-notice/index.tsx#L45-L46

It's part of the `CartCheckoutSidebarCompatibilityNotice`, which is using this TS interface

https://github.com/woocommerce/woocommerce/blob/27899be1549d2d3670ca1e7527a5a88fbbc499c8/plugins/woocommerce-blocks/assets/js/editor-components/sidebar-compatibility-notice/index.tsx#L16-L19

Initially, you mentioned 
> wouldn't it be safer to concatenate "woocommerce/" with the block name an let it error out if something invalid appears here?

What would be your proposal in this case? Something like this:

```ts
const blockName = `woocommerce/${ block }`;
if (
	blockName !== 'woocommerce/cart' &&
	blockName !== 'woocommerce/checkout'
) {
	throw new Error( 'Invalid blockName' );
}
```

Reply:
If we're only accepting 'cart' | 'checkout' earlier, then `const blockName = `woocommerce/${ block }`;` will suffice right? Any other value specified earlier will be rejected at interface level, that if is unneeded an unreachable.

If the interface changes to be less strict, then it would fail later after concatenation, because it wouldn't be able to find blockName.

I don't want to delay this PR, but IMO that would be better than a ternary that defaults the value. Let it fail, imagine in the future you are doing some task, introduce an invalid block at interface level by distraction (typo etc), it defaults to checkout and you never realise it because it will never complain about it.

Reply:
Thanks for clarifying this, @wavvves. I've updated the part you mentioned and went a step ahead. Instead of using
```ts
const blockName = `woocommerce/${ block }`;
```
and
```ts
<SwitchToClassicShortcodeButton
    block={ blockName }
    clientId={ clientId }
    type={ 'generic' }
/>
```
I'm directly calling
```ts
<SwitchToClassicShortcodeButton
    block={ `woocommerce/${ block }` }
    clientId={ clientId }
    type={ 'generic' }
/>

Please let me know what you think about that approach.